lib/tasks/instrumentation_generator/templates/dependency_detection.tt

```diff
@@ -0,0 +1,29 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative '<%= @name.downcase %>/instrumentation'
+require_relative '<%= @name.downcase %>/chain'
+require_relative '<%= @name.downcase %>/prepend'
+
+DependencyDetection.defer do
+  named :<%= @name.downcase %>
+
+  depends_on do
+    # The class that needs to be defined to prepend/chain onto. This can be used
+    # to determine whether the library is installed.
+    defined?(::<%= @name.capitalize %>)
```

Comment:
We might need the equivalent of ActiveSupport's `#classify` here. The agent itself maintains something of a fork of that method that you could use.

```ruby
irb(main):001:0> 'an_example'.capitalize
=> "An_example"
irb(main):002:0> 'an_example'.classify
=> "AnExample"
```

Reply:
I didn't realize the agent had a `#classify` overwrite! Great suggestion! I was contemplating making a `.camelize` dupe, but decided against it and added a TODO instead. I'll prioritize this change.

Reply:
I'm not finding the agent's fork. Could you help point me in the right direction?

Reply:
Ah, I apologize - I got `#constantize` mixed up with `#classify`. `lib/new_relic/language_support.rb` has a fork of `#constantize`.

Here, you want to be able to produce a camel cased constant string without needing to actually be able to retrieve a constant from the constants table (because the instrumented gem is likely not loaded), so `#classify` is better.

We likely don't need the table stuff from `#classify`, so maybe a fork of `#camelize` would suffice.